### DISCUSSION TITLE
Hardware Specification

### DISCUSSION THREAD

**Discussion by psss** [teemtee/tmt#1643] [Ideas]
This discussion is dedicated for announcing new hardware specification topics, issues and pull requests. If you are interested in the area of detailed hardware specification you might want to subscribe to this discussion to get notified about the new stuff available for feedback.

**Comment by psss** [comment]
New hardware specification layout using a dedicated page outlined in #1642. Please, review.

**Comment by psss** [comment]
### tpm

New key `tpm` covering `Trusted Platform Module` proposed in #1644. Please review

**Comment by psss** [comment]
### compatible

New key `compatible` solving `distro` compatibily requested in #1582 proposed in #1645. Please, have a look.

**Comment by juk** [comment]
New cpu property: "flag" is proposed. Please review [#1755](https://github.com/teemtee/tmt/pull/1755)

**Comment by juk** [comment]
A new key: "numa-nodes" to specify the number of NUMA nodes on the system is proposed in [#1756](https://github.com/teemtee/tmt/pull/1756)

**Comment by juk** [comment]
A new key: "model" to provide system's model name is proposed in [#1757](https://github.com/teemtee/tmt/pull/1757)

**Comment by juk** [comment]
A new key: "vendor" to provide system's vendor is proposed in https://github.com/teemtee/tmt/pull/1758

**Comment by juk** [comment]
A new key: "device" with device-id, driver name and vendor-id properties is proposed in https://github.com/teemtee/tmt/pull/1759.

**Reply by happz** [reply]
More points discussed in https://github.com/teemtee/tmt/issues/2337

**Comment by juk** [comment]
A new key: "system" to provide system's attributes such as vendor, model name and NUMA nodes is proposed in https://github.com/teemtee/tmt/pull/1771

**Comment by happz** [comment]
New key: `partitions` for simple custom partitioning of storage devices is proposed in #1835.

**Comment by thrix** [comment]
New key: `kickstart` is being implemented for modifying kickstart installations, see #1282 #1842 for details. Provide feedback.

**Comment by juk** [comment]
New cpu property: "stepping" is proposed. Please review https://github.com/teemtee/tmt/pull/2336

**Comment by happz** [comment]
Requesting specific Beaker job owner has been requested in https://github.com/teemtee/tmt/issues/2345
This can be supported by at least 2 provision plugins, `beaker` and `artemis`, can we make it plugin-agnostic?

**Comment by happz** [comment]
Requesting specific Beaker pool has been requested in https://github.com/teemtee/tmt/issues/2346
This can be supported by at least 2 provision plugins, `beaker` and `artemis`, can we make it plugin-agnostic?

**Comment by happz** [comment]
Requesting specific Beaker system type has been requested in https://github.com/teemtee/tmt/issues/2356
This can be supported by at least 2 provision plugins, `beaker` and `artemis`, can we make it plugin-agnostic?

**Comment by happz** [comment]
Requesting guests with iommu enabled in https://github.com/teemtee/tmt/issues/2392

**Comment by happz** [comment]
Requesting guests with specific GPU in https://github.com/teemtee/tmt/issues/2154

**Comment by happz** [comment]
Requesting specific Beaker lab controller (or avoiding one) has been requested in https://github.com/teemtee/tmt/issues/2457
This can be supported by at least 2 provision plugins, beaker and artemis, can we make it plugin-agnostic?

**Comment by happz** [comment]
Requesting a test for provisioning server with multiple disks with no size constraints in https://github.com/teemtee/tmt/issues/2536

**Comment by happz** [comment]
Allow disk/network HW requirements to speak about the whole set of devices #2721

**Comment by kurik** [comment]
Requesting HW selection based on battery presence: https://github.com/teemtee/tmt/issues/2862

**Comment by psss** [comment]
The `system.management-interface` key is outlined in https://github.com/teemtee/tmt/pull/3459.
